x/concentrated-liquidity/math_test.go

```diff
@@ -180,29 +170,27 @@ func (suite *KeeperTestSuite) TestGetNextSqrtPriceFromAmount1RoundingDown() {
 // sqrtPriceB is the larger of sqrtpCur and the nextPrice
 // calcAmount0Delta = (liquidity * (sqrtPriceB - sqrtPriceA)) / (sqrtPriceB * sqrtPriceA)
 func (suite *KeeperTestSuite) TestCalcAmount0Delta() {
-	testCases := []struct {
-		name            string
+	testCases := map[string]struct {
 		liquidity       sdk.Dec
 		sqrtPCurrent    sdk.Dec
 		sqrtPUpper      sdk.Dec
 		amount0Expected string
 	}{
-		{
-			"happy path",
-			sdk.NewDec(1377927219),
-			sdk.MustNewDecFromStr("70.710678"),
-			sdk.MustNewDecFromStr("74.161984"),
-			"906866",
-			// https://www.wolframalpha.com/input?i=%281377927219+*+%2874.161984+-+70.710678+%29%29+%2F+%2870.710678+*+74.161984%29
+		"happy path": {
+			liquidity:       sdk.MustNewDecFromStr("1517882343.751510418088349649"), // we use the smaller liquidity between liq0 and liq1
+			sqrtPCurrent:    sdk.MustNewDecFromStr("70.710678118654752440"),         // 5000
+			sqrtPUpper:      sdk.MustNewDecFromStr("74.161984870956629487"),         // 5500
+			amount0Expected: "998976.618347426747968399",                            // TODO: should be 998976.618347426388356630
```

Comment:
Please check my previous comment to see if that fixes it here as well

Reply:
As discussed offline, your suggestion fixed 1/2 TODOs, this one was not fixed